smoketests/test.Dockerfile

```diff
@@ -0,0 +1,61 @@
+FROM postgis/postgis:11-2.5
+
+RUN apt-get update \
+ # install newer packages from backports
+ && apt-get install -y --no-install-recommends \
+    g++ \
+    gcc \
+    libc6-dev \
+    make \
+    pkg-config \
+    libgeos-dev \
+    libleveldb-dev \
+    libprotobuf-dev \
+    osmctools \
+    postgis \
+    osmosis \
+ # install postgresql client
+ && apt-get install -y --no-install-recommends \
+      postgresql-client-$PG_MAJOR \
+ && ln -s /usr/lib/libgeos_c.so /usr/lib/libgeos.so \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
+      && apt-get install -y --no-install-recommends git ssh tar gzip ca-certificates curl wget perl postgresql-client \
+      && rm -rf /var/lib/apt/lists/*
+
+RUN curl -LO http://xrl.us/cpanm && chmod +x cpanm \
+    && ./cpanm TAP::Parser::SourceHandler::pgTAP
+
+ENV PERL5LIB /usr/local/lib/perl5
+
+RUN apt-get update \
+      && apt-cache showpkg postgresql-11-pgtap \
+      && apt-get install -y --no-install-recommends postgresql-11-pgtap \
+      && rm -rf /var/lib/apt/lists/*
+
+RUN wget https://golang.org/dl/go1.14.15.linux-amd64.tar.gz \
+    && rm -rf /usr/local/go && tar -C /usr/local -xzf go1.14.15.linux-amd64.tar.gz
+
+ENV GOPATH /usr/local/go
+ENV PATH $PATH:/usr/local/go/bin
+
+ARG IMPOSM_REPO="https://github.com/omniscale/imposm3.git"
+ARG IMPOSM_VERSION="v0.11.1"
+
+# add  github.com/omniscale/imposm3
+RUN mkdir -p $GOPATH/src/github.com/omniscale/imposm3 \
+ && cd  $GOPATH/src/github.com/omniscale/imposm3 \
+ && go get github.com/tools/godep \
+ && git clone --quiet --depth 1 $IMPOSM_REPO -b $IMPOSM_VERSION \
+        $GOPATH/src/github.com/omniscale/imposm3 \
+ && make build \
+ && ( [ -f imposm ] && mv imposm /usr/bin/imposm || mv imposm3 /usr/bin/imposm ) \
```

Comment:
isn't downloading the latest build enough? Why are building from scratch. I mean this is not bad practice but since this is not hitting production maybe it simplifies things.

Reply:
I had a problem combining all needed dependencies for this test to work, first the pre-built binaries didn't work and I've also tried multi-stage without success. I kept the way we already do in the project.